apis/documentai/v1alpha1/processor_types.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	commonv1alpha1 "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/common/v1alpha1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var DocumentAIProcessorGVK = GroupVersion.WithKind("DocumentAIProcessor")
+
+// DocumentAIProcessorSpec defines the desired state of DocumentAIProcessor
+// +kcc:proto=google.cloud.documentai.v1.Processor
+type DocumentAIProcessorSpec struct {
+	commonv1alpha1.CommonSpec `json:",inline"`
+
+	// +required
+	Location string `json:"location"`
+
+	// The DocumentAIProcessor name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// The processor type, such as: `OCR_PROCESSOR`, `INVOICE_PROCESSOR`.
+	//  To get a list of processor types, see
+	//  [FetchProcessorTypes][google.cloud.documentai.v1.DocumentProcessorService.FetchProcessorTypes].
+	// +kcc:proto:field=google.cloud.documentai.v1.Processor.type
+	Type *string `json:"type,omitempty"`
+
+	// The display name of the processor.
+	// +kcc:proto:field=google.cloud.documentai.v1.Processor.display_name
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// The default processor version.
+	// +kcc:proto:field=google.cloud.documentai.v1.Processor.default_processor_version
+	DefaultProcessorVersion *string `json:"defaultProcessorVersion,omitempty"`
```

Comment:
Can we remove this field for now?

Reply:
Done
https://github.com/GoogleCloudPlatform/k8s-config-connector/commit/eb7c57121ffe7d00ef5ae989fa11abf03926d20c